rpcs3/Emu/RSX/RSXThread.cpp

```diff
@@ -316,7 +316,10 @@ namespace rsx
 
 		if (capture_current_frame)
 		{
+			for (const auto &first_count : first_count_commands)
+				vertex_draw_count += first_count.second;
```

Comment:
first_count is a vector of pair (of u32).

RSX draw/draw indexed calls only draw up to 256 vertexes each. So to draw more complex geometry app needs to issue several draw calls between begin/end pause.

For instance you can't draw 1000 vertex at once. You need to split this :
-begin triangles-
Draw 256 vertexes starting at 0
Draw 256 vertexes starting at 256
Draw 256 vertexes starting at 512
Draw 232 vertexes starting at 768
-end-

the first_count vector records the vertex count and offset of each of these lines. There's not vector involved here and D3D12 don't use a staging vector to store vertex or index values ; they're just copied to dx owned memory directly.

I'm storing the various first/count pair because I'm not sure they always represent a contiguous set of vertex. I never encountered case where there was a "gap" though.